openwisp_controller/config/views.py

```diff
@@ -65,43 +79,40 @@ def get_relevant_templates(request, organization_id):
     queryset = (
         Template.objects.filter(**filter_options)
         .filter(org_filters)
+        .order_by("-required", "-default")
         .only("id", "name", "backend", "default", "required")
     )
     selected_templates = []
-    if device_id:
-        selected_templates = (
-            Config.objects.prefetch_related("templates")
-            .only("templates")
-            .get(device_id=device_id)
-            .templates.all()
-        )
-    if group_id:
-        selected_templates = (
-            DeviceGroup.objects.prefetch_related("templates")
-            .only("templates")
-            .get(pk=group_id)
-            .templates.filter(organization_id=organization_id)
-        )
+    if config_id:
+        try:
+            selected_templates = (
+                Config.objects.prefetch_related("templates")
+                .only("templates")
+                .get(pk=config_id)
+                .templates.filter(org_filters)
+                .filter(**filter_options)
+            )
+        except (Config.DoesNotExist, ValueError):
+            pass
 
-    relevant_templates = OrderedDict()
-    for template in selected_templates:
-        relevant_templates[str(template.pk)] = dict(
-            name=template.name,
-            backend=template.get_backend_display(),
-            default=template.default,
-            required=template.required,
-            selected=True,
-        )
+    if group_id:
+        try:
+            selected_templates = (
+                DeviceGroup.objects.prefetch_related("templates")
+                .only("templates")
+                .get(pk=group_id)
+                .templates.filter(org_filters)
+                .filter(**filter_options)
```

Comment:
What's the value of `filter_options` here? Can we add `organization_id=organization_id` here?